.github/workflows/node-zxc-build-release-artifact.yaml

```diff
@@ -115,119 +115,66 @@ permissions:
+      - name: Setup Node.js
+        uses: actions/setup-node@0a44ba7841725637a19e28fa30b79a866c81b0a6 # v4.0.4
         with:
-          gradle-version: ${{ inputs.gradle-version }}
+          node-version: 'lts/*'
 
-      - name: Gradle Update Version (As Specified)
-        uses: gradle/actions/setup-gradle@dbbdc275be76ac10734476cc723d82dfe7ec6eda # v3.4.2
-        if: ${{ inputs.version-policy == 'specified' && !cancelled() && !failure() }}
-        with:
-          gradle-version: ${{ inputs.gradle-version }}
-          arguments: versionAsSpecified -PnewVersion=${{ inputs.new-version }} --scan
+      - name: Install dependencies
+        run: |
+          npm ci
+          npm install --save-dev \
+            @semantic-release/changelog \
+            @semantic-release/git \
+            conventional-changelog-conventionalcommits
 
-      - name: Gradle Update Version (Branch Commit)
-        uses: gradle/actions/setup-gradle@dbbdc275be76ac10734476cc723d82dfe7ec6eda # v3.4.2
-        if: ${{ inputs.version-policy != 'specified' && !cancelled() && !failure() }}
+      - name: Determine Version
+        id: semantic
+        uses: semantic-release/semantic-release@43df51b79ec9c8716b737804486b375459ea177e # v24.1.2
         with:
-          gradle-version: ${{ inputs.gradle-version }}
-          arguments: versionAsPrefixedCommit -PcommitPrefix=${{ steps.parameters.outputs.commit-prefix }} --scan
+          dry_run: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.access-token }}
 
-      - name: Compute Final Effective Version
-        id: effective-version
+      - name: Extract Version Parts
+        id: version-parts
         run: |
-          EFF_VERSION="$(./gradlew showVersion --quiet | tr -d '[:space:]')"
-          PRERELEASE_SUFFIX="$(semver get prerel "${EFF_VERSION}")"
-          PRERELEASE="false"
-          [[ -n "${PRERELEASE_SUFFIX}" ]] && PRERELEASE="true"
-
-          MAJOR_VERSION="$(semver get major "${EFF_VERSION}")"
-          MINOR_VERSION="$(semver get minor "${EFF_VERSION}")"
-
-          echo "number=${EFF_VERSION}" >>"${GITHUB_OUTPUT}"
-          echo "prefix=${MAJOR_VERSION}.${MINOR_VERSION}" >>"${GITHUB_OUTPUT}"
-          echo "prerelease=${PRERELEASE}" >>"${GITHUB_OUTPUT}"
+          VERSION="${{ steps.semantic.outputs.new_release_version }}"
+          echo "major=$(echo $VERSION | cut -d. -f1)" >> $GITHUB_OUTPUT
+          echo "minor=$(echo $VERSION | cut -d. -f2)" >> $GITHUB_OUTPUT
+          echo "patch=$(echo $VERSION | cut -d. -f3 | cut -d- -f1)" >> $GITHUB_OUTPUT
+          echo "prerelease=$(echo $VERSION | grep -o -P '(?<=-).+' || echo '')" >> $GITHUB_OUTPUT
 
-      - name: Cache Build Version
-        uses: actions/cache@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4.1.2
-        with:
-          path: version.txt
-          key: node-build-version-${{ steps.effective-version.outputs.number }}-${{ github.sha }}
```

Comment:
We will need these steps for the branch commit side as well.

Reply:
I've revamped `node-zxc-build-release-artifact.yaml` and these steps are kept as they were.